books/RayTracingTheNextWeek.html

```diff
@@ -187,27 +145,33 @@
     class moving_sphere : public hittable {
       public:
         moving_sphere() {}
-        moving_sphere(
-            point3 ctr0, point3 ctr1, double r, shared_ptr<material> m,
-            double time_start, double time_end)
-            :
-            center0(ctr0), center1(ctr1), radius(r), mat_ptr(m),
-            time0(time_start), time1(time_end)
-        {};
+        moving_sphere(point3 c0, point3 c1, double r, shared_ptr<material> m)
+          : center0(c0), center1(c1), center_vec(c1 - c0), radius(r), mat_ptr(m)
+        {
+            // To Be Implemented
+        };
 
         bool hit(const ray& r, interval ray_t, hit_record& rec) const override {
             // To Be Implemented
```

Comment:
I actually don't like the "to be implemented" and would rather just have all functions that need to be implemented be written plainly, as a `{ }` and go back and denote changes with highlighting, like we've done so far.

Why the change?

Reply:
I was thinking about this. I generally agree that increments are sufficient, and needn't leave hints at "more to come".

For the "hit" method, the listing ends with some additional explanatory text, and then immediately after the full hit function is revealed. It needs to return a `bool`, as well as properly setting the out parameters. I don't like the idea of making the reader parse dummy code, followed by a couple of paragraphs, followed by ripping it out and replacing it with the actual code.

For the constructor, that's just an awkward stopping point during this draft. At line 745, we show how the constructor still has to initialize the cached bounding box of the total possible bounds of the moving sphere. That it happens so much later is kind of crazy, and definitely needs to be fixed.

Note that I used to make these kinds of large changes in two phases: (1) the source code, debugged and working, and (2) all of the remaining text. In this case, I'm trying something a bit new: (1) running and debugged source code, (2) updated listings only, and (3) explanatory text. I'm hoping that this results in fewer changes missed between source and book, and this way I can use the listings as natural prompts for the text. But of course, exposition and development may change in the process (as here).

Looping back to your comment, I often lament the style of these sorts of articles and books. It's so easy to fall into the trap of just dictating the edits, step by step, and leaving no room for trial or experimentation. I kind of left "To Be Implemented" in as a possible prompt for the reader, like "hey, wanna give this a shot?". I don't know how useful or effective this is, and agree that it's clunky as it currently lies. We'll see how it ends up -- on to the text update now.